sql-to-dbsp-compiler/lib/sqllib/src/casts.rs

```diff
@@ -202,74 +202,75 @@ pub fn cast_to_TimeN_time(value: Time) -> Option<Time> {
-        Decimal::zero()
+        SQLDecimal::zero()
     };
     cast_to_decimal_decimal(result, precision, scale)
 }
 
 #[inline]
-pub fn cast_to_decimal_decimal(value: Decimal, _precision: u32, scale: i32) -> Decimal {
+pub fn cast_to_decimal_decimal(value: SQLDecimal, _precision: u32, scale: i32) -> SQLDecimal {
     //value.with_prec(precision as u64).with_scale(scale as i64)
-    let mut result = value;
-    result.rescale(scale as u32);
-    result
+    value.rescale(scale as u32).unwrap()
 }
 
 #[inline]
-pub fn cast_to_decimal_decimalN(value: Option<Decimal>, precision: u32, scale: i32) -> Decimal {
+pub fn cast_to_decimal_decimalN(
+    value: Option<SQLDecimal>,
+    precision: u32,
+    scale: i32,
+) -> SQLDecimal {
     let result = value.unwrap();
     cast_to_decimal_decimal(result, precision, scale)
 }
 
 #[inline]
-pub fn cast_to_decimal_d(value: F64, precision: u32, scale: i32) -> Decimal {
-    let result = Decimal::from_f64(value.into_inner()).unwrap();
+pub fn cast_to_decimal_d(value: F64, precision: u32, scale: i32) -> SQLDecimal {
+    let result = SQLDecimal::from_f64(value.into_inner()).unwrap();
     cast_to_decimal_decimal(result, precision, scale)
 }
 
 #[inline]
-pub fn cast_to_decimal_dN(value: Option<F64>, precision: u32, scale: i32) -> Decimal {
-    let result = Decimal::from_f64(value.unwrap().into_inner()).unwrap();
+pub fn cast_to_decimal_dN(value: Option<F64>, precision: u32, scale: i32) -> SQLDecimal {
+    let result = SQLDecimal::from_f64(value.unwrap().into_inner()).unwrap();
     cast_to_decimal_decimal(result, precision, scale)
 }
 
 #[inline]
-pub fn cast_to_decimal_f(value: F32, precision: u32, scale: i32) -> Decimal {
-    let result = Decimal::from_f32(value.into_inner()).unwrap();
+pub fn cast_to_decimal_f(value: F32, precision: u32, scale: i32) -> SQLDecimal {
+    let result = SQLDecimal::from_f32(value.into_inner()).unwrap();
     cast_to_decimal_decimal(result, precision, scale)
 }
 
 #[inline]
-pub fn cast_to_decimal_fN(value: Option<F32>, precision: u32, scale: i32) -> Decimal {
-    let result = Decimal::from_f32(value.unwrap().into_inner()).unwrap();
+pub fn cast_to_decimal_fN(value: Option<F32>, precision: u32, scale: i32) -> SQLDecimal {
+    let result = SQLDecimal::from_f32(value.unwrap().into_inner()).unwrap();
     cast_to_decimal_decimal(result, precision, scale)
 }
 
 #[inline]
-pub fn cast_to_decimal_s(value: String, precision: u32, scale: i32) -> Decimal {
+pub fn cast_to_decimal_s(value: String, precision: u32, scale: i32) -> SQLDecimal {
     let result = value.trim().parse().unwrap();
-    println!("{}", result);
     cast_to_decimal_decimal(result, precision, scale)
 }
 
 #[inline]
-pub fn cast_to_decimal_sN(value: Option<String>, precision: u32, scale: i32) -> Decimal {
+pub fn cast_to_decimal_sN(value: Option<String>, precision: u32, scale: i32) -> SQLDecimal {
     let result = match value {
-        None => Decimal::zero(),
+        None => SQLDecimal::zero(),
```

Comment:
you probably have to use the specified precision and scale

Reply:
Their precision and scale is set in the line below.